csrc/scheduler/mma_utils.cpp

```diff
@@ -281,6 +281,143 @@ void scheduleWarpTileWithReduction(TensorView* tv, MatMulTileOptions tile) {
   // [Kwo Mwo Nwo Mw Nw Mi Ni Ki]
 }
 
+void scheduleWarpTile(
+    TensorView* tv,
+    MatMulTileOptions tile,
+    const std::vector<MatmulDimRole>& merged_roles) {
+  // Tiles are always given in [M, N, K]
+  auto cta_tile = tile.cta_tile;
+  auto warp_tile = tile.warp_tile;
+  auto instruction_tile = tile.instruction_tile;
+
+  // Do not split K dimension of CTA tile into multiple warp tiles
+  NVF_CHECK(
+      cta_tile.k == warp_tile.k,
+      "CTA tile and warp tile must have same K dimension");
+
+  // Find _inner_ tile dimensions present in tv
+  std::vector<int64_t> inner_dims;
+  std::vector<int64_t> required_inner_dim_sizes;
+  int64_t m_dim = -1, n_dim = -1, k_dim = -1;
+  int64_t min_inner_pos = tv->nDims(); // position of outermost inner dim
+  for (size_t i : c10::irange(merged_roles.size())) {
+    int64_t i_tv = tv->nDims() - 1 - (int64_t)i;
+    int64_t i_roles = (int64_t)merged_roles.size() - 1 - (int64_t)i;
+    if (i_tv < 0 || i_roles < 0) {
+      break;
+    }
+    inner_dims.push_back(i_tv);
+    MatmulDimRole role = merged_roles.at(i_roles);
+    switch (role) {
+      case MatmulDimRole::M:
+        m_dim = i_tv;
+        required_inner_dim_sizes.push_back(cta_tile.m);
+        break;
+      case MatmulDimRole::N:
+        n_dim = i_tv;
+        required_inner_dim_sizes.push_back(cta_tile.n);
+        break;
+      case MatmulDimRole::K:
+        k_dim = i_tv;
+        required_inner_dim_sizes.push_back(cta_tile.k);
+        break;
+      default:
+        continue;
+        break;
+    }
+    min_inner_pos = std::min(min_inner_pos, i_tv);
+  }
+
+  NVF_ERROR(
+      k_dim != -1,
+      "scheduleWarpTile should only be called on an operand with K dimension but found ",
+      tv->toString());
+
+  mma_utils::checkDimSize(tv, inner_dims, required_inner_dim_sizes);
+
+  //     m_dim  n_dim k_dim
+  //[...     M,     N,    K]
+  // Distribute warp tile:
+  if (m_dim != -1) {
```

Comment:
Does `-1` mean "the last dim", or "not found"? We might want to use `std::optional<int64_t> m_dim` for better readability here.

Reply:
Sounds good. It means "not found", but you're right it's more readable to use optionals.